Arabica37

CalciumNitride opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->
A new keyboard that split and column staggered

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/arabica37/arabica37.c:
```diff
@@ -0,0 +1,10 @@
+#include "arabica37.h"
+#include "ssd1306.h"
```
To be blunt, I'd rather not see this implementation of OLED support. 

I'd rather see one of the more generalized ones, such as: https://docs.qmk.fm/#/feature_oled_driver

drashna commented on keyboards/arabica37/ssd1306.c:
```diff
@@ -0,0 +1,344 @@
+#ifdef SSD1306OLED
```
I would really prefer that the OLED Driver or QWIIC OLED feature be used here. 

Or worst case, that the /drivers/avr/ssd1306.c file be used instead, rather than having another copy of this sitting in a keyboard folder.  (this is not a knock against you, at all, but it's better if common code is used, since you don't miss out on fixes and improvements by doing so).

drashna commented:
Well, thank you for the changes that you've made!

That said, it looks like you're using a custom implementation of the split keyboard code, still.   While that may be fine, we'd really rather get everyone onto the "split common" code.  It's features a number of improvements and more functionality, though it does lose some functionality (mirroring the keymaps mostly). 

If you could, could you convert over?  Mostly that involves removing the `i2c`, `serial`, `split_*`, and custom matrix.c files,  and enabling the split functionally via:
https://docs.qmk.fm/#/feature_split_keyboard?id=firmware-configuration

If there is a specific reason that you're using the custom split code, that's fine.  


Otherwise, it looks good!

CalciumNitride commented:
As per directions, I added `SPLIT_KEYBOARD = yes` and removed some files and related code, but I couldn't compile it.
The error message says no rule for making 'keyboards/arabica37/serial.c'.
What do you think could be the cause?

drashna commented:
You may need to run `make clean`.  

Travis CI seems to be happy with it, and compiles fine. And testing this locally, it works fine too

drashna commented on keyboards/arabica37/pro_micro.h:
```diff
@@ -0,0 +1,358 @@
+/*
```
Specifically, this entire file shouldn't be needed, especially if yo've switched to split common.

CalciumNitride commented:
thank you. I could compile without any problem!
then I have uploaded revised files.
Please give me a re review.

noroadsleft commented on keyboards/arabica37/keymaps/default/rules.mk:
```diff
@@ -0,0 +1,25 @@
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
+SWAP_HANDS_ENABLE = no      # Enable one-hand typing
```
None of these rules need to be repeated here. These rules will inherit from `arabica37/rev1/rules.mk` and `arabica37/rules.mk`.

`SWAP_HANDS_ENABLE` defaults to `no` anyway; you don't need to disable it here.

```suggestion
```

noroadsleft commented on keyboards/arabica37/keymaps/jp_default/config.h:
```diff
@@ -0,0 +1,42 @@
+// #define EE_HANDS
+
+#define USE_SERIAL_PD2
+
```
```suggestion
```

noroadsleft commented on keyboards/arabica37/keymaps/jp_default/keymap.c:
```diff
@@ -0,0 +1,198 @@
+  RAISE,
+  ADJUST,
+  BACKLIT,
+  RGBRST
```
```suggestion
  LOWER = SAFE_RANGE,
  RAISE,
  ADJUST,
  RGBRST
```

noroadsleft commented on keyboards/arabica37/keymaps/jp_default/keymap.c:
```diff
@@ -0,0 +1,198 @@
+  //|------+------+------+------+------+------+------|  |------+------+------+------+------+------+------|
+       TRNS,                       TRNS,  TRNS,  TRNS,     TRNS,  TRNS,  TRNS,                       TRNS \
+  //`------------------------------------------------'  `------------------------------------------------'
+  )
```
```suggestion
  [_QWERTY] = LAYOUT_kc( \
  //,-----------------------------------------.         ,-----------------------------------------.
        ESC,     Q,     W,     E,     R,     T,               Y,     U,     I,     O,     P,  BSPC,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
      CTLTB,     A,     S,     D,     F,     G,               H,     J,     K,     L,  MINS,  QUOT,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
      ADJST,     Z,     X,     C,     V,     B,               N,     M,  COMM,   DOT,  SLSH,    RO,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
      SFIME,                      LOWDL, GUIMH,           ALTHN, RAIBS,                       LSFT,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
                                    SPC,                           ENT                             \
  //`-----------------------------------------'         `-----------------------------------------'
  ),

  [_LOWER] = LAYOUT_kc( \
  //,-----------------------------------------.         ,-----------------------------------------.
       TRNS,  PSCR,  SCRS,   INS,  CAPS,  TRNS,            MAIL,  WSCH,  TRNS,  TRNS,  TRNS,  TRNS,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,  MUTE,  LEFT,    UP, RIGHT,  PGUP,              F1,    F2,    F3,    F4,    F5,    F6,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,   DTL,  HOME,  DOWN,   END,  PGDN,              F7,    F8,    F9,   F10,   F11,   F12,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,                       TRNS,  TRNS,            TRNS,  TRNS,                       TRNS,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
                                   TRNS,                          TRNS                             \
  //`-----------------------------------------'         `-----------------------------------------'
  ),

  [_RAISE] = LAYOUT_kc( \
  //,-----------------------------------------.         ,-----------------------------------------.
       TRNS,  EXLM,    AT,  HASH,   DLR,  PERC,            PDOT,     7,     8,     9,  PPLS,  TRNS,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,  RBRC,  LABK,  ASTR,  SCLN,  LBRC,            PSLS,     4,     5,     6,  PMNS,  PDOT,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,  BSLS,  RABK,  LPRN,  JYEN,  UNDS,            PAST,     1,     2,     3,     0,  COMM,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,                       TRNS,  TRNS,            TRNS,  TRNS,                       TRNS,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
                                   TRNS,                          TRNS                             \
  //`-----------------------------------------'         `-----------------------------------------'
  ),

  [_ADJUST] = LAYOUT_kc( \
  //,-----------------------------------------.         ,-----------------------------------------.
       TRNS,  LRST,  LTOG,  LMOD, RESET, XXXXX,           XXXXX, XXXXX, XXXXX, XXXXX, XXXXX, XXXXX,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,  LHUI,  LSAI,  LVAI,  SWAP, XXXXX,           XXXXX, XXXXX, XXXXX, XXXXX, XXXXX, XXXXX,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,  LHUD,  LSAD,  LVAD,  NORM, XXXXX,           XXXXX, XXXXX, XXXXX, XXXXX, XXXXX, XXXXX,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
       TRNS,                       TRNS,  TRNS,            TRNS,  TRNS,                       TRNS,\
  //|------+------+------+------+------+------|         |------+------+------+------+------+------|
                                   TRNS,                          TRNS                             \
  //`-----------------------------------------'         `-----------------------------------------'
  )
```

noroadsleft commented on keyboards/arabica37/keymaps/jp_default/rules.mk:
```diff
@@ -0,0 +1,25 @@
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+SWAP_HANDS_ENABLE = no      # Enable one-hand typing
```
```suggestion
```

noroadsleft commented on keyboards/arabica37/keymaps/jp_default/rules.mk:
```diff
@@ -0,0 +1,25 @@
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight.
+SWAP_HANDS_ENABLE = no      # Enable one-hand typing
+OLED_DRIVER_ENABLE = no
```
Any particular reason you have OLED enabled for the default keymap, but disabled for this one?

noroadsleft commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,45 @@
+//rgb led driver
+#include "ws2812.h"
+#endif
+
```
`RGBLIGHT_ENABLE = yes` in `rules.mk` includes this file automatically.

```suggestion
```

noroadsleft commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,45 @@
+  L00, L01, L02, L03, L04, L05,           R00, R01, R02, R03, R04, R05, \
+  L10, L11, L12, L13, L14, L15,           R10, R11, R12, R13, R14, R15, \
+  L20, L21, L22, L23, L24, L25,           R20, R21, R22, R23, R24, R25, \
+  L26,                L30, L31, L32, R26, R30, R31,                R32 \
```
```suggestion
  L00, L01, L02, L03, L04, L05,           R00, R01, R02, R03, R04, R05, \
  L10, L11, L12, L13, L14, L15,           R10, R11, R12, R13, R14, R15, \
  L20, L21, L22, L23, L24, L25,           R20, R21, R22, R23, R24, R25, \
  L26,                L31, L32,           R26, R30,                R32, \
                        L30,                 R31                        \
```

noroadsleft commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,45 @@
+    KC_##L00, KC_##L01, KC_##L02, KC_##L03, KC_##L04, KC_##L05,                     KC_##R00, KC_##R01, KC_##R02, KC_##R03, KC_##R04, KC_##R05, \
+    KC_##L10, KC_##L11, KC_##L12, KC_##L13, KC_##L14, KC_##L15,                     KC_##R10, KC_##R11, KC_##R12, KC_##R13, KC_##R14, KC_##R15, \
+    KC_##L20, KC_##L21, KC_##L22, KC_##L23, KC_##L24, KC_##L25,                     KC_##R20, KC_##R21, KC_##R22, KC_##R23, KC_##R24, KC_##R25, \
+    KC_##L26,                               KC_##L30, KC_##L31, KC_##L32, KC_##R26, KC_##R30, KC_##R31,                               KC_##R32 \
```
```suggestion
  L00, L01, L02, L03, L04, L05,           R00, R01, R02, R03, R04, R05, \
  L10, L11, L12, L13, L14, L15,           R10, R11, R12, R13, R14, R15, \
  L20, L21, L22, L23, L24, L25,           R20, R21, R22, R23, R24, R25, \
  L26,                L31, L32,           R26, R30,                R32, \
                        L30,                 R31                        \
  ) \
  LAYOUT( \
    KC_##L00, KC_##L01, KC_##L02, KC_##L03, KC_##L04, KC_##L05,           KC_##R00, KC_##R01, KC_##R02, KC_##R03, KC_##R04, KC_##R05, \
    KC_##L10, KC_##L11, KC_##L12, KC_##L13, KC_##L14, KC_##L15,           KC_##R10, KC_##R11, KC_##R12, KC_##R13, KC_##R14, KC_##R15, \
    KC_##L20, KC_##L21, KC_##L22, KC_##L23, KC_##L24, KC_##L25,           KC_##R20, KC_##R21, KC_##R22, KC_##R23, KC_##R24, KC_##R25, \
    KC_##L26,                               KC_##L31, KC_##L32,           KC_##R26, KC_##R30,                               KC_##R32, \
                                            KC_##L30,                               KC_##R31                                          \
```

noroadsleft commented on keyboards/arabica37/readme.md:
```diff
@@ -0,0 +1,16 @@
+# Arabica3/7
+
+
+![arabica3/7](https://user-images.githubusercontent.com/46911478/76915532-9c333080-68f8-11ea-863a-8a7899d1d9c4.jpg)
```
```suggestion
![arabica3/7](https://user-images.githubusercontent.com/46911478/76915532-9c333080-68f8-11ea-863a-8a7899d1d9c4.jpg)
```

noroadsleft commented on keyboards/arabica37/rev1/config.h:
```diff
@@ -0,0 +1,83 @@
+#define PRODUCT_ID      0x3060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    CalciumNitride
+#define PRODUCT         Arabica37
```
```suggestion
#define PRODUCT         Arabica3/7
```

noroadsleft commented on keyboards/arabica37/rules.mk:
```diff
@@ -0,0 +1,72 @@
+# CFLAGS += -flto
+
+# MCU name
+#MCU = at90usb1287
```
I'd actually much rather have all these rules set in `keyboards/arabica37/rev1/rules.mk`, rather than here.

```suggestion
# MCU name
```

noroadsleft commented on keyboards/arabica37/rules.mk:
```diff
@@ -0,0 +1,72 @@
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
```
QMK sets all of these based on `MCU = atmega32u4` above; they're not needed here at all unless you need to change them.

```suggestion
```

noroadsleft commented on keyboards/arabica37/rules.mk:
```diff
@@ -0,0 +1,72 @@
+# Bootloader
+#     This definition is optional, and if your keyboard supports multiple bootloaders of
+#     different sizes, comment this out, and the correct address will be loaded
+#     automatically (+60). See bootloader.mk for all options.
```
```suggestion
# Bootloader selection
#   Teensy       halfkay
#   Pro Micro    caterina
#   Atmel DFU    atmel-dfu
#   LUFA DFU     lufa-dfu
#   QMK DFU      qmk-dfu
#   ATmega32A    bootloadHID
#   ATmega328P   USBasp
```

noroadsleft commented on keyboards/arabica37/rules.mk:
```diff
@@ -0,0 +1,72 @@
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
```
QMK sets this also.

```suggestion
```

noroadsleft commented on keyboards/arabica37/rules.mk:
```diff
@@ -0,0 +1,72 @@
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
```suggestion
#   change yes to no to disable
```

noroadsleft commented on keyboards/arabica37/rules.mk:
```diff
@@ -0,0 +1,72 @@
+SUBPROJECT_rev1 = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Aligns this part to the QMK template for this file (`quantum/template/avr/rules.mk`). No logic changes here.

```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = no        # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

noroadsleft commented on keyboards/arabica37/readme.md:
```diff
@@ -0,0 +1,16 @@
+
+Keyboard Maintainer: [CalciumNitride](https://github.com/CalciumNitride)  
+Hardware Supported: Arabica37 PCB, Pro Micro  
+Hardware Availability: [PCB & Case Data](https://github.com/calciumnitride/arabica37)
```
```suggestion
* Keyboard Maintainer: [CalciumNitride](https://github.com/CalciumNitride)
* Hardware Supported: Arabica3/7 PCB, Pro Micro
* Hardware Availability: [PCB & Case Data](https://github.com/calciumnitride/arabica37)
```

noroadsleft commented on keyboards/arabica37/readme.md:
```diff
@@ -0,0 +1,16 @@
+
+Make example for this keyboard (after setting up your build environment):
+
+    make arabica37:default
```
```suggestion
    make arabica37/rev1:default
```

noroadsleft commented on keyboards/arabica37/info.json:
```diff
@@ -0,0 +1,62 @@
+{
```
Realized your file was copied from the Corne. Try this:

```json
{
    "keyboard_name": "arabica37 rev. 1",
    "url": "",
    "maintainer": "qmk",
    "width": 15.25,
    "height": 5.75,
    "layouts": {
        "LAYOUT": {
            "layout": [
                {"label":"Esc", "x":0.25, "y":0.75},
                {"label":"Q", "x":1.25, "y":0.75},
                {"label":"W", "x":2.25, "y":0.25},
                {"label":"E", "x":3.25, "y":0},
                {"label":"R", "x":4.25, "y":0.25},
                {"label":"T", "x":5.25, "y":0.5},
                
                {"label":"Y", "x":9, "y":0.5},
                {"label":"U", "x":10, "y":0.25},
                {"label":"I", "x":11, "y":0},
                {"label":"O", "x":12, "y":0.25},
                {"label":"P", "x":13, "y":0.75},
                {"label":"Backspace", "x":14, "y":0.75},
                
                {"label":"Ctrl", "x":0.25, "y":1.75},
                {"label":"A", "x":1.25, "y":1.75},
                {"label":"S", "x":2.25, "y":1.25},
                {"label":"D", "x":3.25, "y":1},
                {"label":"F", "x":4.25, "y":1.25},
                {"label":"G", "x":5.25, "y":1.5},
                
                {"label":"H", "x":9, "y":1.5},
                {"label":"J", "x":10, "y":1.25},
                {"label":"K", "x":11, "y":1},
                {"label":"L", "x":12, "y":1.25},
                {"label":";", "x":13, "y":1.75},
                {"label":"'", "x":14, "y":1.75},
                
                {"label":"Adjust", "x":0.25, "y":2.75},
                {"label":"Z", "x":1.25, "y":2.75},
                {"label":"X", "x":2.25, "y":2.25},
                {"label":"C", "x":3.25, "y":2},
                {"label":"V", "x":4.25, "y":2.25},
                {"label":"B", "x":5.25, "y":2.5},
                
                {"label":"N", "x":9, "y":2.5},
                {"label":"M", "x":10, "y":2.25},
                {"label":",", "x":11, "y":2},
                {"label":".", "x":12, "y":2.25},
                {"label":"/", "x":13, "y":2.75},
                {"label":"\\", "x":14, "y":2.75},
                
                {"label":"Shift", "x":0, "y":3.75, "w":2.25},
                {"label":"Lower", "x":4.375, "y":3.75, "w":1.5},
                {"label":"GUI", "x":5.875, "y":3.75, "w":1.25},
                
                {"label":"Alt", "x":8.125, "y":3.75, "w":1.25},
                {"label":"Raise", "x":9.375, "y":3.75, "w":1.5},
                {"label":"Shift", "x":13, "y":3.75, "w":2.25},
                
                {"label":"Space", "x":4.875, "y":4.75, "w":1.25},
                {"label":"Enter", "x":8.625, "y":4.75, "w":1.25}
            ]
        }
    }
}
```

![image](https://user-images.githubusercontent.com/18669334/77486238-6c9a9180-6dec-11ea-8e88-bdffaafba03f.png)

CalciumNitride commented:
Thanks for your review.
the changes added.

noroadsleft commented on keyboards/arabica37/rev1/rules.mk:
```diff
@@ -0,0 +1,37 @@
+
+
+
+# Bootloader selection
```
```suggestion
# Bootloader selection
```

noroadsleft commented on keyboards/arabica37/rev1/rules.mk:
```diff
@@ -0,0 +1,37 @@
+
+SPLIT_KEYBOARD = yes
+
+DEFAULT_FOLDER = arabica37/rev1
```
We're already here. :)

```suggestion
```

noroadsleft commented on keyboards/arabica37/keymaps/default/rules.mk:
```diff
@@ -0,0 +1,7 @@
+OLED_DRIVER_ENABLE = yes
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
This inherits from the rev1 `rules.mk`, it's not needed here.

```suggestion
```

noroadsleft commented on keyboards/arabica37/keymaps/jp_default/rules.mk:
```diff
@@ -0,0 +1,7 @@
+OLED_DRIVER_ENABLE = yes
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
```suggestion
```

fauxpark commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,43 @@
+
+#include "../arabica37.h"
+
+//void promicro_bootloader_jmp(bool program);
```
```suggestion
```

fauxpark commented on keyboards/arabica37/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,173 @@
+ //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
+    KC_TRNS,                        KC_TRNS,KC_TRNS,           KC_TRNS,KC_TRNS,                        KC_TRNS,\
+ //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
+                                    KC_TRNS,                           KC_TRNS                                 \
```
```suggestion
  [_QWERTY] = LAYOUT(
 //,-----------------------------------------------.          ,-----------------------------------------------.
     KC_ESC,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,              KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,KC_BSPC,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_LCTL,   KC_A,   KC_S,   KC_D,   KC_F,   KC_G,              KC_H,   KC_J,   KC_K,   KC_L,KC_SCLN,KC_QUOT,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
     ADJUST,   KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,              KC_N,   KC_M,KC_COMM, KC_DOT,KC_SLSH,KC_BSLS,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_LSFT,                          LOWER,KC_LGUI,           KC_LALT,  RAISE,                        KC_RSFT,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
                                     KC_SPC,                            KC_ENT
 //`-----------------------------------------------'          `-----------------------------------------------'
 ),
 [_LOWER] = LAYOUT(
 //,-----------------------------------------------.          ,-----------------------------------------------.
    KC_TRNS,   KC_1,   KC_2,   KC_3,   KC_4,   KC_5,              KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_TRNS,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,KC_HOME,KC_LEFT,  KC_UP,KC_RGHT,KC_PGUP,             KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS, KC_END,KC_PSCR,KC_DOWN, KC_INS,KC_PGDN,             KC_F7,  KC_F8,  KC_F9, KC_F10, KC_F11, KC_F12,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,                        KC_TRNS,KC_TRNS,           KC_TRNS,KC_TRNS,                        KC_TRNS,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
                                    KC_TRNS,                           KC_TRNS
 //`-----------------------------------------------'          `-----------------------------------------------'
 ),
 [_RAISE] = LAYOUT(
 //,-----------------------------------------------.          ,-----------------------------------------------.
    KC_TRNS,KC_EXLM,  KC_AT,KC_HASH, KC_DLR,KC_PERC,           KC_CIRC,KC_PSLS,KC_PAST,KC_PMNS,KC_PPLS,KC_TRNS,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,KC_RBRC,KC_LABK,KC_ASTR,KC_SCLN,KC_LBRC,              KC_1,   KC_2,   KC_3,   KC_4,   KC_5, KC_EQL,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,KC_BSLS,KC_RABK,KC_LPRN,KC_JYEN,KC_UNDS,              KC_6,   KC_7,   KC_8,   KC_9,   KC_0,KC_PENT,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+--- ---+-------+-------|
    KC_TRNS,                        KC_TRNS,KC_TRNS,           KC_TRNS,KC_TRNS,                        KC_TRNS,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+--- ---+-------+-------|
                                    KC_TRNS,                           KC_TRNS
 //`-----------------------------------------------'          `-----------------------------------------------'
 ),
 [_ADJUST] = LAYOUT(
 //,-----------------------------------------------.          ,-----------------------------------------------.
    KC_TRNS, RGBRST,RGB_TOG,  KC_NO,  KC_NO,  KC_NO,             KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,RGB_HUI,RGB_SAI,RGB_VAI,AG_SWAP,  KC_NO,             KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,RGB_HUD,RGB_SAD,RGB_VAD,AG_NORM,  KC_NO,             KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,  KC_NO,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
    KC_TRNS,                        KC_TRNS,KC_TRNS,           KC_TRNS,KC_TRNS,                        KC_TRNS,
 //|-------+-------+-------+-------+-------+-------|          |-------+-------+-------+-------+-------+-------|
                                    KC_TRNS,                           KC_TRNS
```

fauxpark commented on keyboards/arabica37/keymaps/jp_default/keymap.c:
```diff
@@ -0,0 +1,201 @@
+#define KC_RESET RESET
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_QWERTY] = LAYOUT_kc( \
```
Layout macros that automatically prepend `KC_` are discouraged, as they do not play well with the Configurator, and make some of the more advanced keycodes such as `MO()` and `LT()` harder to use, since you need to manually define aliases beginning with `KC_`.

CalciumNitride replied:
The default keymap doesn't use "KC_" macro, but I want to keep means of use it as another keymap. Should I still remove this?

drashna replied:
It's not THE `default` keymap, but it's still A default keymap.

fauxpark commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,43 @@
+   KC_##L20, KC_##L21, KC_##L22, KC_##L23, KC_##L24, KC_##L25,           KC_##R20, KC_##R21, KC_##R22, KC_##R23, KC_##R24, KC_##R25, \
+   KC_##L26,                               KC_##L31, KC_##L32,           KC_##R26, KC_##R30,                               KC_##R32, \
+                                           KC_##L30,                               KC_##R31                                          \
+)
```
```suggestion
```

noroadsleft commented on keyboards/arabica37/rev1/rules.mk:
```diff
@@ -0,0 +1,37 @@
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```
I made a mistake here before, giving you a suggestion that turned this off.

```suggestion
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
```

drashna commented on keyboards/arabica37/rev1/rev1.c:
```diff
@@ -0,0 +1,27 @@
+      	_delay_ms(150);
+      	stop_all_notes();
+    #endif
+}
```
This entire block should be removed.  this is in quantum, so, you're basically overriding the behavior in a way that can't be easily fixed. 

```suggestion
```

drashna commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include "../arabica37.h"
```
```suggestion
#include "arabica37.h"
```

fauxpark commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,42 @@
+#define has_usb() is_keyboard_master()
+uint8_t is_master;
+
+//void promicro_bootloader_jmp(bool program);
```
```suggestion
```

drashna commented on keyboards/arabica37/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,173 @@
+
+int RGB_current_mode;
+
+void matrix_init_user(void) {
```
This really should be: 
```suggestion
void keyboard_post_init_user(void) {
```

That way, it has the properly loaded info.

drashna commented on keyboards/arabica37/keymaps/jp_default/keymap.c:
```diff
@@ -0,0 +1,201 @@
+
+int RGB_current_mode;
+
+void matrix_init_user(void) {
```
```suggestion
void keyboard_post_init_user(void) {
```

zvecr commented on keyboards/arabica37/rev1/rev1.h:
```diff
@@ -0,0 +1,41 @@
+#include "arabica37.h"
+
+#include "quantum.h"
+
+#define has_usb() is_keyboard_master()
+uint8_t is_master;
+
```
```suggestion

```

zvecr commented on keyboards/arabica37/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,173 @@
+#include QMK_KEYBOARD_H
+
+extern uint8_t is_master;
+
```
```suggestion

```

zvecr commented on keyboards/arabica37/keymaps/jp_default/keymap.c:
```diff
@@ -0,0 +1,201 @@
+#include QMK_KEYBOARD_H
+
+extern uint8_t is_master;
+
```
```suggestion

```

noroadsleft commented on keyboards/arabica37/info.json:
```diff
@@ -0,0 +1,65 @@
+                {"label":"Raise", "x":9.375, "y":3.75, "w":1.5},
+                {"label":"Shift", "x":13, "y":3.75, "w":2.25},
+
+                {"label":"Space", "x":4.875, "y":4.75, "w":1.25},
+                {"label":"Enter", "x":8.625, "y":4.75, "w":1.25}
```
I messed up the positioning on my suggestion; this fixes it.

```suggestion
                {"label":"Space", "x":5.125, "y":4.75, "w":1.25},
                {"label":"Enter", "x":8.875, "y":4.75, "w":1.25}
```

CalciumNitride commented:
@noroadsleft thanks! I fixed.

drashna commented:
the `pro_micro.h` file needs deleting, still. 

`git rm keyboards/arabica37/pro_micro.h` and a commit should get rid of it, I believe.

CalciumNitride commented:
@drashna thanks.  It was removed.

drashna commented on keyboards/arabica37/keymaps/default/glcdfont.c:
```diff
@@ -0,0 +1,232 @@
+// This is the 'classic' fixed-space bitmap font for Adafruit_GFX since 1.0.
```
Is there a reason for having this in the default keymap folder, as opposed to the keyboard's folder? 

Mostly, because in it's current location, it's likely to get duplicated repeatedly. 

If you could, please move it to keyboard folder.

CalciumNitride replied:
I moved it. Is this ok?
